src/vos/evt_iter.c

```diff
@@ -358,7 +382,16 @@ evt_iter_probe_sorted(struct evt_context *tcx, struct evt_iterator *iter,
 	index = evt_iter_probe_find(iter, rect);
 	if (index == -1)
 		return -DER_NONEXIST;
+
 	iter->it_index = index;
+	entry = evt_ent_array_get(iter->it_entries, index);
+	if (iter->it_forward)
+		entry->en_sel_ext.ex_lo = max(rect->rc_ex.ex_lo, entry->en_ext.ex_lo);
```

Comment:
This seems weird.   Why isn't it already fixed up when selecting the rectangle to begin with?

Also, can the low and high both be wrong (e.g. one rectangle in the tree with smaller selection ?)

Reply:
> This seems weird. Why isn't it already fixed up when selecting the rectangle to begin with?
> 

Hmm, I thought evt_iter_probe_sorted is selected the started rectangle for iteration.  So we either fix the started rectangle here in this allocated array, or fix during iter_fetch,  but this seems clearer and simpler here. Or I miss the question?

> Also, can the low and high both be wrong (e.g. one rectangle in the tree with smaller selection ?)

hmm, I thought evt_iter_probe_find() will make sure the index is located either on the overlapped extent or the extent closet with the rect anchor.   And this update will make sure the following iteration starting  

1. if no overlap extent is found, then start from the closet extent in the array.
2. if there is overlap extent,  then start from the anchor extent.

Reply:
I don't see why it's adjusted as "max(anchor.ex_lo, recx.ex_lo)", is it possible that "recx.ex_lo > anchor.ex_lo"? If so, you'll partially (or fully) skip anchor, if recx.ex_lo is always greater than anchor.ex_lo, you'd just put an assert and use anchor.ex_lo here.

Is this iterator probe used only for rebuild? Or you want it being used by DAOS recx enumeration API as well? If it'll be used by DAOS recx enumeration API as well, I have few questions:

1. Will you change the recx enumeration to use sorted evtree iterator in follow-on patch?
2. What's the semantic of recx enumeration API will be? I mean what's the expected behavior if the anchor is merged with new updates, or even fully deleted by aggregation? How will application deal with such situation? I think you'd clarify all these in the commit comment.

I personally think that we didn't clearly define the recx enumeration API, if we still want to support (and improve it), we'd clearly define the expected behavior so that application can have a definite expectation.